### PR TITLE
Upgrade Axum to 0.7, tower to 0.5, opentelemetry to 0.23, and hyper to 1.0

### PR DESCRIPTION
**Pull request by jeckhart**
# Description

Upgrade to axum 0.7 and upgrade associated deps as the same time. It seemed like it wouldn't be possible to upgrade axum without also upgrading these other deps based on new deps within the axum stack.

I also fixed several rust fmt concerns as well as a few other warnings, lints, and other issues in order to get the github project to build cleanly after creation. 

~~I haven't tackled wasm yet and will pick that up next.~~
Update: Added changes to support the rust+wasm tree as well.

## Link to issue

Please add a link to any relevant issues/tickets.

## Type of change

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

Axum 0.7 definitely has breaking changes, so this would certainly not be an inplace upgrade for projects. However, since this is a template that shouldn't be a concern here.

## Test plan (required)

Demonstrate the code is solid. Are changes to the templates tested in the top-level project workflows?

Which commands did you test with and what are the expected results? Which tests have you added or updated? Do the
tests cover all of the changes included in this PR?

I ran all cargo tests, lints, fmt checks, benches, etc. that the project contains. I kept iterating until this would run in Github cleanly.

### CODE REVIEW

**Comment by zeeshanlakhani** [comment]
@jeckhart I'll review this, this week. If you re-push, CI should re-run as well (and may need some updates).

**Comment by jeckhart** [comment]
Excellent, I added changes to upgrade both `rust` and `rust+wasm` workspaces as well as additional changes to fix linting errors.

File: rust+wasm/.github/workflows/tests_and_checks.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
           - stable
           - nightly
           # minimum version
-          - 1.67
+          - "1.75"
```
Comment (zeeshanlakhani) [inline rust+wasm/.github/workflows/tests_and_checks.yml:24]: do we need the quotes here?

**Review by zeeshanlakhani — APPROVED** [review]
minor comment, otherwise, great work @jeckhart. Thank you for this.

File: rust+wasm/.github/workflows/tests_and_checks.yml (hunk 1 above)
Comment (jeckhart) [reply rust+wasm/.github/workflows/tests_and_checks.yml:24]: We don't need them for 1.75, however, we would need them for `1.70`, or `1.80` as the yaml parser will interpret those as numbers and shorten them to `1.7` and `1.8` respectively. I saw this happen when testing and one of the libs needed `1.70` and the action installed `1.7` instead.

**Pull request merged**